### ISSUE TITLE
Differing arguments in abstract methods not being picked up in dunder methods

### ISSUE THREAD

**Issue by oliversheridanmethven** [PyCQA/pylint#11295]
### Bug description

Adapted from the example from arguments differ, the derived class only raises W0221 arguments-differ for the non dunder methods. I would expect the violations for the following dunder methods to also complain:

```python
from abc import ABC, abstractmethod

class Base(ABC):
    @abstractmethod
    def __init__(self, foo, bar):
        """Some initializer"""
    
    def __enter__(self):
        return self
    
    @abstractmethod
    def __call__(self, foo, bar):
        """Some abstract method"""
    
    @abstractmethod
    def __exit__(self, exc_type, exc_value, traceback):
        pass
    
    @abstractmethod
    def abstract(self, foo, bar):
        """Some abstract method"""
    
    def non_abstract(self, foo, bar):
        """Some non abstract method"""


class Derived(Base):
    def __init__(self, foo, bar, baz):
        """Some initializer"""
        
    def abstract(self, foo, bar, baz):
        """Some abstract method"""
    
    def non_abstract(self, foo, bar, baz):
        """Some non abstract method"""
        
    def __exit__(self, *args):
        pass

    def __call__(self, foo, bar, baz):
        """Some abstract method"""
```

**EDIT** - Removed unintended `@abstractmethod` in `Derived`. 

### Configuration

```ini

```

### Command used

```shell
pylint example.py
```

### Pylint output

```python
************* Module misc
misc.py:7:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:10:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:14:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:18:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:22:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:30:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:33:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:36:0: C0303: Trailing whitespace (trailing-whitespace)
misc.py:43:0: C0304: Final newline missing (missing-final-newline)
misc.py:1:0: C0114: Missing module docstring (missing-module-docstring)
misc.py:3:0: C0115: Missing class docstring (missing-class-docstring)
misc.py:5:23: C0104: Disallowed name "foo" (disallowed-name)
misc.py:5:28: C0104: Disallowed name "bar" (disallowed-name)
misc.py:12:23: C0104: Disallowed name "foo" (disallowed-name)
misc.py:12:28: C0104: Disallowed name "bar" (disallowed-name)
misc.py:20:23: C0104: Disallowed name "foo" (disallowed-name)
misc.py:20:28: C0104: Disallowed name "bar" (disallowed-name)
misc.py:23:27: C0104: Disallowed name "foo" (disallowed-name)
misc.py:23:32: C0104: Disallowed name "bar" (disallowed-name)
misc.py:27:0: C0115: Missing class docstring (missing-class-docstring)
misc.py:31:4: W0221: Number of parameters was 3 in 'Base.abstract' and is now 4 in overriding 'Derived.abstract' method (arguments-differ)
misc.py:34:4: W0221: Number of parameters was 3 in 'Base.non_abstract' and is now 4 in overriding 'Derived.non_abstract' method (arguments-differ)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

All derived methods mentioned should complain. 

### Pylint version

```shell
4.0.7
```

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Look like it's due to f214ab129 from issue #1042. I think we do have a false negative on ``__call__`` but ``__init__`` could be by design as it's not substitutable.

**Comment by oliversheridanmethven** [comment]
> but `__init__` could be by design as it's not substitutable.

@Pierre-Sassoulas - For this instance, my intent was that the `__init__` method must implement the same signature. 

I am not sure I see the distinction you're making between `__call__` and `__init__`.
